classes/Twig/Block/ChannelInfoBlock.php

```diff
@@ -79,7 +84,130 @@ public function render()
+                'currentFormattedPhpVersion' => $this->getFormattedVersion(PHP_VERSION),
+                'targetFormattedPSVersion' => $this->getFormattedVersion($upgradeInfo['version_num'], self::PS_VERSION_DISPLAY_MAX_PRECISION),
             ]
         );
     }
+
+    /**
+     * Builds array of formatted data for the compatibility table display
+     *
+     * @return array
+     */
+    public function buildCompatibilityTableDisplay()
+    {
+        $startPrestaShopVersion = $previousPHPRange = $previousPrestaVersion = null;
+        $result = [];
+        $i = 0;
+        foreach (UpgradeSelfCheck::PHP_PS_VERSIONS as $prestashopVersion => $phpVersions) {
+            ++$i;
+            if (is_null($startPrestaShopVersion)) {
+                $startPrestaShopVersion = $prestashopVersion;
+                $previousPHPRange = $phpVersions;
+            }
+
+            $isCurrentPrestaVersion = $this->isCurrentPrestashopVersion($startPrestaShopVersion);
+            if ($phpVersions === $previousPHPRange) {
+                $previousPrestaVersion = $prestashopVersion;
+            } else {
+                $label = $this->buildPSLabel($startPrestaShopVersion, $previousPrestaVersion);
+                $result[$label]['php_versions'] = $this->buildPhpVersionsList($previousPHPRange);
+                $result[$label]['is_current'] = $isCurrentPrestaVersion;
+                $startPrestaShopVersion = $prestashopVersion;
+                $result[$label]['is_target'] = $this->getFormattedVersion($this->channelInfo->getInfo()['version_num'], self::PS_VERSION_DISPLAY_MAX_PRECISION) === $label;
+                $previousPrestaVersion = null;
+            }
+            if ($i === count(UpgradeSelfCheck::PHP_PS_VERSIONS)) {
+                $result[$prestashopVersion]['php_versions'] = $this->buildPhpVersionsList($phpVersions);
+                $result[$prestashopVersion]['is_current'] = $isCurrentPrestaVersion;
+                $result[$prestashopVersion]['is_target'] = $this->getFormattedVersion($this->channelInfo->getInfo()['version_num'], self::PS_VERSION_DISPLAY_MAX_PRECISION) === $prestashopVersion;
+            }
+            $previousPHPRange = $phpVersions;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Builds PrestaShop version label for display
+     *
+     * @param string $startVersion
+     * @param string $endVersion
+     *
+     * @return string
+     */
+    public function buildPSLabel($startVersion, $endVersion)
+    {
+        if ($startVersion === self::PS_MINIMAL_VERSION) {
+            return '>= ' . self::PS_MINIMAL_VERSION;
+        }
+
+        return $startVersion .= $endVersion ? ' ~ ' . $endVersion : '';
+    }
+
+    /**
+     * Builds a list of php versions for a given php version range
+     *
+     * @param array $phpVersionRange
+     *
+     * @return array
+     */
+    public function buildPhpVersionsList($phpVersionRange)
+    {
+        $phpStart = $phpVersionRange[0];
```

Comment:
Since it's a public method, maybe correctly control what's inside the array? Or put it private/protected.